### PR TITLE
Merge 0.22 -> 0.23

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

File: client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import org.http4s.implicits._
+import org.http4s.multipart.Multipart
+import org.http4s.multipart.Part
+import org.typelevel.ci._
+
+import java.util.Arrays
+import java.util.Locale
+import scala.concurrent.duration._
+
+abstract class ClientRouteTestBattery(name: String) extends Http4sSuite with Http4sClientDsl[IO] {
```
Comment (danicheg) [inline client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala:36]: If I'm not wrong, this is the `0.22`-related suite.

File: ember-client/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+import org.http4s.ember.client.internal.ClientHelpers
+import org.http4s.headers.`User-Agent`
+import org.typelevel.keypool._
+import org.typelevel.log4cats.Logger
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration._
+
+final class EmberClientBuilder[F[_]: Concurrent: Timer: ContextShift] private (
```
Comment (danicheg) [inline ember-client/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala:38]: CE2 related things came from 0.22

File: client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala (hunk 1 above)
Comment (armanbilge) [reply client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala:36]: Yes, these files are actually misplaced.
```
[error] (coreJS / Test / test) Misplaced sources in /home/runner/work/http4s/http4s/core/src
[error] (testingJS / Test / test) Misplaced sources in /home/runner/work/http4s/http4s/testing/src
[error] (testsJS / Test / test) Misplaced sources in /home/runner/work/http4s/http4s/tests/src
[error] (ember-clientJS / Test / test) Misplaced sources in /home/runner/work/http4s/http4s/ember-client/src
[error] (clientJS / Test / test) Misplaced sources in /home/runner/work/http4s/http4s/client/src
```

Comment (rossabaker) [reply client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala:36]: Ah, hell.  I think git didn't catch all the moved files, and I didn't catch the ones it didn't catch.

File: ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala [hunk 3]
```diff
@@ -374,17 +379,17 @@ object EmberClientBuilder extends EmberClientBuilderCompanionPlatform {
...
       100
     }
-    val maxTotal = 100
-    val idleTimeInPool = 30.seconds // 30 Seconds in Nanos
-    val additionalSocketOptions = List.empty[SocketOption]
-    val userAgent = Some(
+    val maxTotal: Int = 100
+    val idleTimeInPool: FiniteDuration = 30.seconds // 30 Seconds in Nanos
+    val additionalSocketOptions: List[SocketOption] = List.empty[SocketOption]
+    val userAgent: Option[`User-Agent`] = Some(
```
Comment (armanbilge) [inline ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala:392]: MiMa is complaining about this one, but only on Scala 2 I think (it wants `Some`). So idk if we can put a type annotation that works across all Scala versions.

File: ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala [hunk 4]
```diff
@@ -389,7 +389,7 @@ object EmberClientBuilder extends EmberClientBuilderCompanionPlatform {
     val maxTotal: Int = 100
     val idleTimeInPool: FiniteDuration = 30.seconds // 30 Seconds in Nanos
     val additionalSocketOptions: List[SocketOption] = List.empty[SocketOption]
-    val userAgent: Option[`User-Agent`] = Some(
+    val userAgent: Some[`User-Agent`] = Some(
```
Comment (armanbilge) [inline ember-client/shared/src/main/scala/org/http4s/ember/client/EmberClientBuilder.scala:392]: Wait, hang on, it looks like this worked? This makes me concerned about what's going on in Scala 3 MiMa which is passing before and after this change ...

**Review by armanbilge — CHANGES_REQUESTED** [review]
Sorry, hang on until I understand what's going on here.

**Review by armanbilge — APPROVED** [review]
Oh, don't mind me. I see now that it's all private. Really strange that Scala 3 recognized it though and Scala 2 couldn't, usually it's the other way around 🤔

**Pull request merged**